packages/components/src/slot-fill/types.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * External dependencies
+ */
+import type { Component, MutableRefObject, ReactNode, RefObject } from 'react';
+
+export type DistributiveOmit< T, K extends keyof any > = T extends any
+	? Omit< T, K >
+	: never;
+
+export type SlotKey = string | symbol;
+
+export type FillProps = any;
```

Comment:
Let's use something slightly more defined, maybe 

```suggestion
export type FillProps = Record< string, any >;
``` ?

Reply:
It would be better to use Record type.